[Docs] Translate embed/go/memory.md to zh

Bevisy opened the pull request:
translate embed/go/memory.md to zh

Fix #1091

Signed-off-by: Binbin Zhang <binbin36520@gmail.com>

Bevisy commented:
PTAL @alabulei1 @MileyFu

alabulei1 commented on docs/book/zh/src/embed/go/memory.md:
```diff
@@ -1 +1,647 @@
+# 将复杂参数传递给 Wasm 函数
+
+WebAssembly 规范的一个问题是它只支持非常有限的数据类型。如果要嵌入具有复杂调用参数或返回值的 WebAssembly 函数，则需要同时在 Go SDK 和 WebAssembly 函数端管理内存指针。
+此类复杂的调用参数和返回值包括字符串和字节数组等动态内存结构。
```
3 4 5 都是一行。

alabulei1 commented on docs/book/zh/src/embed/go/memory.md:
```diff
@@ -1 +1,647 @@
+在[此示例](https://github.com/second-state/WasmEdge-go-examples/tree/master/go_MemoryGreet)中，我们将演示如何从 Go 应用程序调用[基于 Rust 的 WebAssembly 函数](https://github.com/second-state/WasmEdge-go-examples/blob/master/go_MemoryGreet/rust_memory_greet/src/lib.rs)。
+特别是，我们将讨论如何传递字符串数据。
+
+> 在 WebAssembly 中向 Rust 函数传递和返回复杂值的另一种方法是使用`wasm-bindgen`编译器工具。你可以[在这里了解更多](bindgen.md).
```
代码块前后空格

alabulei1 commented on docs/book/zh/src/embed/go/memory.md:
````diff
@@ -1 +1,647 @@
+# 输出的 WASM 将是 `target/wasm32-wasi/debug/rust_memory_greet_lib.wasm`.
+```
+
+[Go SDK 应用程序](https://github.com/second-state/WasmEdge-go-examples/blob/master/go_MemoryGreet/greet_memory.go)必须从 WasmEdge 虚拟机中调用`allocate`以获得一个指向字符串参数的指针。
````
61 62 63 都是一行，然后代码块前后空格

alabulei1 commented on docs/book/zh/src/embed/go/memory.md:
```diff
@@ -1 +1,647 @@
+
+TinyGo 函数获取字符串的内存指针，并自己构造 TinyGo 字符串。
+
+> 编译 WebAssembly 程序需要`main()`为空，才能正确设置 WASI。
```
代码块前后空格

alabulei1 commented on docs/book/zh/src/embed/go/memory.md:
```diff
@@ -1 +1,647 @@
+  subject := "WasmEdge"
+  lengthOfSubject := len(subject)
+
+  // Allocate memory for the subject, and get a pointer to it.
```
这些注释也翻译一下吧

alabulei1 commented on docs/book/zh/src/embed/go/memory.md:
````diff
@@ -1 +1,647 @@
+tinygo build -o greet.wasm -target wasi greet.go
+```
+
+[Go SDK 应用程序](https://github.com/second-state/WasmEdge-go-examples/blob/master/go_MemoryGreetTinyGo/greet_memory.go)必须从 WasmEdge 虚拟机中调用`malloc`以获得一个指向字符串参数的指针。
````
205 206 207 应该是一行吧，代码块要空格哈

alabulei1 commented on docs/book/zh/src/embed/go/memory.md:
```diff
@@ -1 +1,647 @@
+  subject := "WasmEdge"
+  lengthOfSubject := len(subject)
+
+  // Allocate memory for the subject, and get a pointer to it.
```
注释也翻译一下

alabulei1 commented on docs/book/zh/src/embed/go/memory.md:
```diff
@@ -1 +1,647 @@
+
+在[此示例](https://github.com/second-state/WasmEdge-go-examples/tree/master/go_AccessMemoryTinyGo)中，我们将演示如何调用[基于 TinyGo 的 WebAssembly 函数](https://github.com/second-state/WasmEdge-go-examples/blob/master/go_AccessMemoryTinyGo/fib.go)以及将数组传入和传出 Go 应用程序。
+
+函数 "fibArray "接收一个数组作为调用参数，并将其填入
```
这应该都是一行

alabulei1 replied:
fibArray 这里应该是代码块吧

alabulei1 reviewed (commented):
Thanks for your contribution! There are two things we should take care of.
1. Add a space between code blocks.
2. Translate the noes in the code blocks.

Bevisy commented:
> Thanks for your contribution! There are two things we should take care of.
> 
> 1. Add a space between code blocks.
> 2. Translate the noes in the code blocks.

@alabulei1 Thanks for your review, I have reworked the translation according to the review comments, and added all the code comments to the translation.

alabulei1 commented:
@Bevisy  LGTM! Only one place needs to be changed. Please check out.

Bevisy commented:
> @Bevisy LGTM! Only one place needs to be changed. Please check out.

@alabulei1 Thanks, but can you mark where there are still changes to be made?
<!-- (Because I wanted to try a longer translation, I picked up this translation task in the translation task list.) -->

alabulei1 commented:
> > @Bevisy LGTM! Only one place needs to be changed. Please check out.
> 
> @alabulei1 Thanks, but can you mark where there are still changes to be made?

@Bevisy Sure. 第246行，英文字母前后空一格

Bevisy commented:
> > > @Bevisy LGTM! Only one place needs to be changed. Please check out.
> > 
> > 
> > @alabulei1 Thanks, but can you mark where there are still changes to be made?
> 
> @Bevisy Sure. 第246行，英文字母前后空一格

Sorry actually forgot the `NULL` word in the comment, has been fixed now, thanks a lot.

alabulei1 commented:
> > > > @Bevisy LGTM! Only one place needs to be changed. Please check out.
> > > 
> > > 
> > > @alabulei1 Thanks, but can you mark where there are still changes to be made?
> > 
> > 
> > @Bevisy Sure. 第246行，英文字母前后空一格
> 
> Sorry actually forgot the `NULL` word in the comment, it's fixed now. Thanks a lot.

@Bevisy please also add a space between `NULL` in lines 237, 104, 95. Sorry for missing.

Bevisy commented:
> > > > > @Bevisy LGTM! Only one place needs to be changed. Please check out.
> > > > 
> > > > 
> > > > @alabulei1 Thanks, but can you mark where there are still changes to be made?
> > > 
> > > 
> > > @Bevisy Sure. 第246行，英文字母前后空一格
> > 
> > 
> > Sorry actually forgot the `NULL` word in the comment, it's fixed now. Thanks a lot.
> 
> @Bevisy please also add a space between `NULL` in lines 237, 104, 95. Sorry for missing them.

Done 😄

Bevisy commented:
@alabulei1 重新修改了标点符号，将部分句号“。”修改成冒号“：”，分别为 line 3、50、140等共13处，规范标点符号使用。
参考：https://teachinghand.com/chinese-punctuation-guide/。

PTAL @hydai

alabulei1 commented:
LGTM! @Bevisy Thanks for your contribution.

@hydai  Could you help merge this PR?

hydai commented on docs/book/zh/src/embed/go/memory.md:
````diff
@@ -1 +1,632 @@
+# 输出的 WASM 将是 target/wasm32-wasi/debug/rust_access_memory_lib.wasm.
+```
+
+[Go SDK 应用程序](https://github.com/second-state/WasmEdge-go-examples/blob/master/go_AccessMemory/run.go)必须从 WasmEdge 虚拟机中调用 `allocate` ，以获得一个指向数组的指针。然后它将调用Rust中的`fib_array()`函数并传入指针。在函数返回后，Go应用程序将使用 WasmEdge 的 `store` API，从调用参数（ `fib_array()` ）或返回值（ `fib_array_return_memory()` ）中的指针构建一个数组。Go 应用最终会调用 `deallocate` 来释放内存空间。
````
中英文間應有空格隔開

Bevisy replied:
Thanks, fixed now.

Bevisy commented:
If there is nothing else that needs to be modified, I hope to be able to merge it as soon as possible, thank you! @hydai
